client/client.go

```diff
@@ -272,13 +286,19 @@ func (c *Client) Close() error {
 func (c *Client) connectStreams(chans <-chan ssh.NewChannel) {
 	for ch := range chans {
 		remote := string(ch.ExtraData())
+		socks := remote == "socks"
```

Comment:
This line needs this https://github.com/aus/chisel/blob/client-socks/server/handler.go#L166-L170  check afterwards

Reply:
Thanks for the feedback! I guess I'm a bit confused. My understanding of the linked code block is that it essentially checks whether the `--socks5` server option was enabled. Are you saying that the `--socks5` server option should be checked and enabled in order for client socks remotes `R:socks` to work? Or are you saying there should be a client option of `--socks5`? Or neither? 😄 

In my original commit, I had added the client option `--socks5` to enable the client socks server. However after your suggestions, I've concluded that the client `--socks5` option is redundant if the client is also specifying socks in the reverse remote `R:socks`.

I understand the security reasoning for having the server `--socks5` option. Since the client specifies the remote, the server should not allow clients with free network access unless explicitly enabled.  But for client socks, I don't really see any risk where neither the server nor client would need to enable the `--socks5` option. And the `--reverse` server option already permits or denies the server-side risk.

Hopefully you can clarify things for me. Thanks!